### DISCUSSION TITLE
How to "Rerun the Homebrew installer under Rosetta 2."?

### DISCUSSION THREAD

**Discussion by oguzbilgic** [Homebrew/discussions#153] [Everyday usage] [answered]
Running `/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"` on the new apple m1 macbook brew outputs the following: 

```
Homebrew is not (yet) supported on ARM processors!
Rerun the Homebrew installer under Rosetta 2.
If you really know what you are doing and are prepared for a very broken experience you can use another installation option for installing on ARM:
  https://docs.brew.sh/Installation
```

My question is, how can I run the install command using `rosetta 2` as the output suggests.

**Comment by Rylan12** [comment, marked as answer]
There’s a [PR in progress](https://github.com/Homebrew/brew/pull/9126/files#diff-e181586945bce9183b7c22ca539f9dd72d289287f6a763fe90667800614ddaf0R53) to explain exactly that in more detail.

From the PR (referring to installing a formula):

> To use it, you can choose between:
> 
> * running your Terminal from Rosetta 2
> * running `arch -x86_64 brew` instead of `brew`.

To install Homebrew itself, you can run `arch -x86_64 /bin/bash` instead of `/bin/bash`.
To configure Terminal to run under Rosetta 2, select Terminal.app in Finder, hit ⌘I, select _Open with Rosetta_, then restart your Terminal.

_I stole this response from Claudia's answer in https://github.com/Homebrew/brew/issues/9173#issuecomment-729215356_

**Reply by oguzbilgic** [reply]
Thanks @Rylan12 

Honestly, I feel like just manually cloning the repo is better, I'm hoping that it compiles everything else for m1 properly, instead of using x86 for everything that will be installed using brew

**Reply by SMillerDev** [reply]
A lot of things are broken on ARM native. You're really better off installing under Rosetta.

**Reply by tyron** [reply]
@SMillerDev do you know if there will be a clear update path if we start using it under Rosetta and then Homebrew becomes supported natively in ARM? Moreover, will the update path include not only Homebrew itself, but also any formulae installed that made the switch and can run natively?

Not trying to push, just trying to understand if start doing things right now is a good option for the average user or if we really should be waiting until further notice. Thanks!

**Reply by SMillerDev** [reply]
Ruby runs on ARM so Homebrew itself is fine, the problem is that we don't have CI so we can't really test and fix formulae. Until we can do that and bottle everything we'll declare ARM supported and you can use the normal installer again.

**Reply by ajaygautam** [reply]
```
% arch -x86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)" 
arch: posix_spawnp: /bin/bash: Bad CPU type in executable
```

Any ideas?

**Reply by shiningshisa** [reply]
> ```
> % arch -x86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)" 
> arch: posix_spawnp: /bin/bash: Bad CPU type in executable
> ```
> 
> 
> Any ideas?

The selected answer in this question suggests that you need to install Rosetta 2. Unsure if this is actually your issue though, I don't recall explicitly installing Rosetta but seem to have Homebrew working [answer](https://stackoverflow.com/questions/64882584/how-to-run-the-homebrew-installer-under-rosetta-2-on-m1-macbook)

**Reply by ajaygautam** [reply]
Found the problem of "Bad CPU type in executable" - You are right Rosetta 2 was not installed.
(This was one of the first things I tried to install on the machine and ran into this issue)

*Solution*: Launch a program, that has a UI, and needs Rosetta 2... That will auto prompt you to install Rosetta.
As of Dec 10, 2020 XCode needs Rosetta... so launch it and it should ask to install Rosetta.
If that does not work... pick any program, right click, select Get Info, select the "Open using Rosetta" checkbox.
Now launch the program... this will trigger Rosetta 2 install.
Install Rosetta 2, and revert the Checkbox.
Now relaunch `arch -x86_64 ...`

Thanks

**Reply by carlocab** [reply]
@ajaygautam Did

    softwareupdate --install-rosetta

from the [linked answer above](https://stackoverflow.com/a/64883440) not work?

**Reply by ajaygautam** [reply]
@carlocab - my bad. I must have over looked that (remember getting a little impatient with this).

**Comment by flypenguin** [comment]
I get the exact same error message when using `arch -x86_64` or when running terminal under rosetta. tried both, about 150 times in all thinkable combinations.

what am I doing wrong?!

```bash
#  arch -x86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
Password:
Homebrew is not (yet) supported on ARM processors!
Rerun the Homebrew installer under Rosetta 2.
If you really know what you are doing and are prepared for a very broken
experience you can use another installation option for installing on ARM:
  https://docs.brew.sh/Installation
```

**Reply by SMillerDev** [reply]
Try enabling Rosetta for the terminal app and running it as such.

**Reply by carlocab** [reply]
You may need to install Rosetta first, if you haven’t already:

     softwareupdate --install-rosetta

**Reply by flypenguin** [reply]
tried both about 20 times. no change. i really don't get it, I seem to be the only one with that issue.

**Reply by mcnarik** [reply]
The answer above is correct, you have to go to the "Get info" for the Terminal app in the Applications folder and setup the: Open using Rosetta. After that you have to reopen the Terminal app. This part is tricky because you really have to close the Terminal completely. I use the Command + Q shortcut to be sure. When done, you can run the installation without problems.
